clickhouse_send_query.go

```diff
@@ -1,22 +1,30 @@
 package clickhouse
 
 import (
+	"context"
 	"github.com/ClickHouse/clickhouse-go/lib/data"
 	"github.com/ClickHouse/clickhouse-go/lib/protocol"
 )
 
-func (ch *clickhouse) sendQuery(query string, externalTables []ExternalTable) error {
+func (ch *clickhouse) sendQuery(ctx context.Context, query string, externalTables []ExternalTable) error {
 	ch.logf("[send query] %s", query)
 	if err := ch.encoder.Uvarint(protocol.ClientQuery); err != nil {
 		return err
 	}
-	if err := ch.encoder.String(""); err != nil {
+	var queryId string
+	queryIdValue := ctx.Value("query_id")
```

Comment:
It is not recommended to use context keys as plain strings. It may cause collisions with other packages.
Better to do something like this
```golang
type key uint8
const keyQueryID key

// wrap query_id into context
func WithQueryID(ctx context.Context, queryID string) context.Context {
    return context.WithValue(ctx, keyQueryID, queryID)
}

func (ch *clickhouse) sendQuery(...) {
    ...
    queryIdValue := ctx.Value(keyQueryID)
    ...
}
```
Mode details in golang blog https://blog.golang.org/context#TOC_3.2.

Reply:
This is a good point. I change it according to your suggestion. 
@kshvakov I'm not sure if it's ok to put the WithQueryID in stmt.go. Please help to find the best place or if you have a better idea about this.